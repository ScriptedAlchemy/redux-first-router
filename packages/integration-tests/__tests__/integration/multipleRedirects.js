import createTest from '../../__helpers__/createTest'

createTest('multiple redirects are honored', {
  SECOND: {
    path: '/second',
    beforeEnter: ({ dispatch }) => dispatch({ type: 'REDIRECTED' }),
    thunk() {},
  },
  REDIRECTED: {
    path: '/redirected',
    beforeEnter: ({ dispatch }) => dispatch({ type: 'REDIRECTED_AGAIN' }),
    thunk() {},
  },
  REDIRECTED_AGAIN: {
    path: '/redirected-again',
    thunk() {},
  },
})

createTest('multiple redirects are honored after enter', {
  SECOND: {
    path: '/second',
    thunk: ({ dispatch }) => dispatch({ type: 'REDIRECTED_AFTER' }),
    onComplete() {},
  },
  REDIRECTED_AFTER: {
    path: '/redirected-after',
    thunk: ({ dispatch }) => dispatch({ type: 'REDIRECTED_AGAIN_AFTER' }),
    onComplete() {},
  },
  REDIRECTED_AGAIN_AFTER: {
    path: '/redirected-again-after',
    onComplete() {},
  },
})
