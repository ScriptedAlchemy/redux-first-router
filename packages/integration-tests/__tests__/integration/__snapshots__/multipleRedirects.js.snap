// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multiple redirects are honored after enter: 1 - response 1`] = `
Object {
  "basename": "",
  "hash": "",
  "location": Object {
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/first",
          "scene": "",
          "search": "",
          "url": "/first",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/redirected-again-after",
          "scene": "",
          "search": "",
          "url": "/redirected-again-after",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "REDIRECTED_AGAIN_AFTER",
      },
    ],
    "from": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 1,
        "key": "345678",
        "pathname": "/second",
        "scene": "",
        "search": "",
        "url": "/second",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "SECOND",
    },
    "index": 1,
    "key": "345678",
    "kind": "replace",
    "length": 2,
    "n": 1,
    "pathname": "/redirected-again-after",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 0,
        "key": "345678",
        "pathname": "/first",
        "scene": "",
        "search": "",
        "url": "/first",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "FIRST",
    },
    "scene": "",
    "search": "",
    "status": 302,
    "url": "/redirected-again-after",
  },
  "params": Object {},
  "query": Object {},
  "state": Object {},
  "type": "REDIRECTED_AGAIN_AFTER",
}
`;

exports[`multiple redirects are honored after enter: 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`multiple redirects are honored after enter: 1 - routes - REDIRECTED - onComplete 1`] = `0`;

exports[`multiple redirects are honored after enter: 1 - routes - REDIRECTED_AFTER - onComplete 1`] = `0`;

exports[`multiple redirects are honored after enter: 1 - routes - REDIRECTED_AFTER - thunk 1`] = `1`;

exports[`multiple redirects are honored after enter: 1 - routes - REDIRECTED_AGAIN_AFTER - onComplete 1`] = `1`;

exports[`multiple redirects are honored after enter: 1 - routes - SECOND - onComplete 1`] = `0`;

exports[`multiple redirects are honored after enter: 1 - routes - SECOND - thunk 1`] = `1`;

exports[`multiple redirects are honored after enter: 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/first",
          "scene": "",
          "search": "",
          "url": "/first",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/redirected-again-after",
          "scene": "",
          "search": "",
          "url": "/redirected-again-after",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "REDIRECTED_AGAIN_AFTER",
      },
    ],
    "from": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 1,
        "key": "345678",
        "pathname": "/second",
        "scene": "",
        "search": "",
        "url": "/second",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "SECOND",
    },
    "hash": "",
    "index": 1,
    "key": "345678",
    "kind": "replace",
    "length": 2,
    "n": 1,
    "params": Object {},
    "pathname": "/redirected-again-after",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 0,
        "key": "345678",
        "pathname": "/first",
        "scene": "",
        "search": "",
        "url": "/first",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "FIRST",
    },
    "query": Object {},
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 302,
    "type": "REDIRECTED_AGAIN_AFTER",
    "universal": false,
    "url": "/redirected-again-after",
  },
  "title": "REDIRECTED_AGAIN_AFTER",
}
`;

exports[`multiple redirects are honored after enter: 1 - title 1`] = `"REDIRECTED_AGAIN_AFTER"`;

exports[`multiple redirects are honored: 1 - response 1`] = `
Object {
  "basename": "",
  "hash": "",
  "location": Object {
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/first",
          "scene": "",
          "search": "",
          "url": "/first",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/redirected-again",
          "scene": "",
          "search": "",
          "url": "/redirected-again",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "REDIRECTED_AGAIN",
      },
    ],
    "from": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 1,
        "key": "345678",
        "pathname": "/second",
        "scene": "",
        "search": "",
        "url": "/second",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "SECOND",
    },
    "index": 1,
    "key": "345678",
    "kind": "replace",
    "length": 2,
    "n": 1,
    "pathname": "/redirected-again",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 0,
        "key": "345678",
        "pathname": "/first",
        "scene": "",
        "search": "",
        "url": "/first",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "FIRST",
    },
    "scene": "",
    "search": "",
    "status": 302,
    "url": "/redirected-again",
  },
  "params": Object {},
  "query": Object {},
  "state": Object {},
  "type": "REDIRECTED_AGAIN",
}
`;

exports[`multiple redirects are honored: 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`multiple redirects are honored: 1 - routes - REDIRECTED - beforeEnter 1`] = `1`;

exports[`multiple redirects are honored: 1 - routes - REDIRECTED - thunk 1`] = `0`;

exports[`multiple redirects are honored: 1 - routes - REDIRECTED_AGAIN - thunk 1`] = `1`;

exports[`multiple redirects are honored: 1 - routes - SECOND - beforeEnter 1`] = `1`;

exports[`multiple redirects are honored: 1 - routes - SECOND - thunk 1`] = `0`;

exports[`multiple redirects are honored: 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/first",
          "scene": "",
          "search": "",
          "url": "/first",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/redirected-again",
          "scene": "",
          "search": "",
          "url": "/redirected-again",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "REDIRECTED_AGAIN",
      },
    ],
    "from": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 1,
        "key": "345678",
        "pathname": "/second",
        "scene": "",
        "search": "",
        "url": "/second",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "SECOND",
    },
    "hash": "",
    "index": 1,
    "key": "345678",
    "kind": "replace",
    "length": 2,
    "n": 1,
    "params": Object {},
    "pathname": "/redirected-again",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 0,
        "key": "345678",
        "pathname": "/first",
        "scene": "",
        "search": "",
        "url": "/first",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "FIRST",
    },
    "query": Object {},
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 302,
    "type": "REDIRECTED_AGAIN",
    "universal": false,
    "url": "/redirected-again",
  },
  "title": "REDIRECTED_AGAIN",
}
`;

exports[`multiple redirects are honored: 1 - title 1`] = `"REDIRECTED_AGAIN"`;
